### PR TITLE
Fix k8schain ECR support on EKS.

### PR DESCRIPTION
**Pull request by mattmoor**
This fixes the k8schain support for ECR on EKS by updating Vincent's credentialprovider fork to 1.17.

There were some interface changes, which alterred k8schain.go, but otherwise things just seem to work now:
```
$ KO_DOCKER_REPO=912237561242.dkr.ecr.us-west-2.amazonaws.com ko apply -BRf pkg/authn/k8schain/tests/
2020/03/13 08:39:29 Using base gcr.io/distroless/static:latest for github.com/google/go-containerregistry/pkg/authn/k8schain/tests/explicit
2020/03/13 08:39:29 Using base gcr.io/distroless/static:latest for github.com/google/go-containerregistry/pkg/authn/k8schain/tests/serviceaccount
2020/03/13 08:39:29 Using base gcr.io/distroless/static:latest for github.com/google/go-containerregistry/pkg/authn/k8schain/tests/noauth
2020/03/13 08:39:29 Using base gcr.io/distroless/static:latest for github.com/google/go-containerregistry/pkg/authn/k8schain/tests/implicit
2020/03/13 08:39:30 Building github.com/google/go-containerregistry/pkg/authn/k8schain/tests/serviceaccount
2020/03/13 08:39:30 Building github.com/google/go-containerregistry/pkg/authn/k8schain/tests/noauth
2020/03/13 08:39:30 Building github.com/google/go-containerregistry/pkg/authn/k8schain/tests/explicit
2020/03/13 08:39:30 Building github.com/google/go-containerregistry/pkg/authn/k8schain/tests/implicit
2020/03/13 08:39:36 Publishing 912237561242.dkr.ecr.us-west-2.amazonaws.com/serviceaccount:latest
2020/03/13 08:39:36 Publishing 912237561242.dkr.ecr.us-west-2.amazonaws.com/implicit:latest
2020/03/13 08:39:36 Publishing 912237561242.dkr.ecr.us-west-2.amazonaws.com/noauth:latest
2020/03/13 08:39:36 Publishing 912237561242.dkr.ecr.us-west-2.amazonaws.com/explicit:latest
2020/03/13 08:39:36 Published 912237561242.dkr.ecr.us-west-2.amazonaws.com/serviceaccount@sha256:2b8c7fa8e017b9b0c818eb14ca38cd26e903729b41be0d369c9907b4f56a99bb
2020/03/13 08:39:36 Published 912237561242.dkr.ecr.us-west-2.amazonaws.com/noauth@sha256:15d837c793602882b5b26f023332f76bf4d52397aceb5de95aabbae13f7325e9
2020/03/13 08:39:36 Published 912237561242.dkr.ecr.us-west-2.amazonaws.com/implicit@sha256:b6a5f54a98af699c3422f72f1a261ebf9ddff112f54619923aee6f207ba3998b
2020/03/13 08:39:36 Published 912237561242.dkr.ecr.us-west-2.amazonaws.com/explicit@sha256:39ece75c033efbb86a6f076ca9f7d1e1dd03e73e2d2fc1b2c22bf9e660d2468e
namespace/explicit-namespace created
secret/explicit-secret created
serviceaccount/explicit created
clusterrolebinding.rbac.authorization.k8s.io/explicit created
pod/explicit created
serviceaccount/implicit created
clusterrolebinding.rbac.authorization.k8s.io/implicit created
pod/implicit created
serviceaccount/noauth created
clusterrolebinding.rbac.authorization.k8s.io/noauth created
pod/noauth created
namespace/serviceaccount-namespace created
secret/serviceaccount-secret created
serviceaccount/serviceaccount created
serviceaccount/serviceaccount created
clusterrolebinding.rbac.authorization.k8s.io/serviceaccount created
pod/serviceaccount created
```

All the test pods complete successfully:
```
NAMESPACE     NAME                      READY   STATUS      RESTARTS   AGE
default       explicit                  0/1     Completed   0          115s
default       implicit                  0/1     Completed   0          115s
default       noauth                    0/1     Completed   0          115s
default       serviceaccount            0/1     Completed   0          115s
```

Previously `implicit` would fail, partially because the ECR credential provider wasn't being registered.

### CODE REVIEW

**Comment by mattmoor** [comment]
DO NOT MERGE this until I can remove the replace in `go.mod` which needs: https://github.com/vdemeester/k8s-pkg-credentialprovider/pull/1 to merge.

File: pkg/authn/k8schain/k8schain.go [hunk 1]
```diff
@@ -119,11 +121,18 @@ func NewNoClient() (authn.Keychain, error) {
...
+type lazyProvider struct {
+	kc    *keychain
+	image string
+}
 
 // Authorization implements Authenticator.
 func (lp lazyProvider) Authorization() (*authn.AuthConfig, error) {
-	authConfig := credentialprovider.LazyProvide(credentialprovider.LazyAuthConfiguration(lp))
+	creds, found := lp.kc.keyring.Lookup(lp.image)
+	if !found || len(creds) < 1 {
```
Comment (vagababov) [inline pkg/authn/k8schain/k8schain.go]: Is there a UT, by chance? :)

Comment (mattmoor) [reply pkg/authn/k8schain/k8schain.go]: Unfortunately the way the K8s credentialprovider package works it is incredibly hard to unit test this stuff, which is why the `ko`-based integration tests in the `test` sub-directory were originally written.

**Comment by mattmoor** [comment]
Alright, this is ready to go, the linked PR has merged, this has been updated, and the integration tests revalidated on ECR.

**Pull request merged**